profiles/killbill/src/test/java/org/killbill/billing/server/log/obfuscators/TestPatternObfuscator.java

```diff
@@ -17,16 +17,112 @@
 
 package org.killbill.billing.server.log.obfuscators;
 
-import org.killbill.billing.server.log.ServerTestSuiteNoDB;
+import java.util.Collection;
+import java.util.List;
+
 import org.mockito.Mockito;
 import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 import ch.qos.logback.classic.spi.ILoggingEvent;
 
-public class TestPatternObfuscator extends ServerTestSuiteNoDB {
+public class TestPatternObfuscator {
+
+    private PatternObfuscator obfuscator;
+
+    @BeforeClass(groups = "fast")
+    public void beforeClass() throws Exception {
+        System.setProperty("killbill.server.log.obfuscate.keywords", "req.requestURI,accountnumber,authenticationdata,banknumber,bic,cardvalidationnum,cavv,ccFirstName,ccLastName,ccNumber,ccTrackData,ccVerificationValue,ccvv,cvNumber,cvc,cvv,email,iban,name,number,password,xid,cell_phone");
+        System.setProperty("killbill.server.log.obfuscate.patterns.separator", "#");
+        System.setProperty("killbill.server.log.obfuscate.patterns", "\\s*=\\s*'([^']+)'#\\s*=\\s*\\\"([^\\\"]+)#\\s*=\\s*([^ '\\\",]+)#\\s*:\\s*'([^'\\\",{}]+)'");
+        obfuscator = new PatternObfuscator();
+    }
+
+    @Test
+    public void testLoadKeywords() {
+        //without setting the killbill.server.log.obfuscate.patterns property
+        System.clearProperty("killbill.server.log.obfuscate.keywords");
+        Collection<String> keywords = obfuscator.loadKeywords();
+        Assert.assertEquals(keywords.size(), 22);
+        Collection<String> actual = List.of(
+                "accountnumber",
+                "authenticationdata",
+                "bankaccountnumber",
+                "banknumber",
+                "bic",
+                "cardvalidationnum",
+                "cavv",
+                "ccFirstName",
+                "ccLastName",
+                "ccNumber",
+                "ccTrackData",
+                "ccVerificationValue",
+                "ccvv",
+                "cvNumber",
+                "cvc",
+                "cvv",
+                "email",
+                "iban",
+                "name",
+                "number",
+                "password",
+                "xid");
+        Assert.assertEquals(actual, keywords);
+
+        //after setting the the killbill.server.log.obfuscate.patterns property
+        System.setProperty("killbill.server.log.obfuscate.keywords", "accountnumber,authenticationdata");
```

Comment:
Actually, should we even have an `@AfterMethod` to clean up system properties?

Reply:
Done